src/cg_ftoc.c

```diff
@@ -197,7 +197,7 @@ CGNSDLL void FMNAME(cg_configure_f, CG_CONFIGURE_F) (cgint_f *what,
     *ier = (cgint_f)cg_configure((int)*what, &C_comm);
 #endif
   } else {
-    *ier = (cgint_f)cg_configure((int)*what, value);
+    *ier = (cgint_f)cg_configure((int)*what, (void *)(*(size_t *)value));
```

Comment:
This change segfaults

Reply:
When does it segfaults ? with which option ?

Reply:
#0  0x000000000043ea20 in cg_set_path (path=0x2f612f747365742f <error: Cannot access memory at address 0x2f612f747365742f>) at cgnslib.c:839
#1  0x0000000000406f3c in cg_configure_f (what=what@entry=0x8cf3c4, value=value@entry=0x7fffffffaf80, ier=ier@entry=0x7fffffffac24) at cg_ftoc.c:209
#2  0x00000000004058ef in write_cgns_1 () at cgwrite_f03.F90:359
#3  0x000000000040424f in main (argc=<optimized out>, argv=<optimized out>) at cgwrite_f03.F90:32
#4  0x000015555388534a in __libc_start_main () from /lib64/libc.so.6
#5  0x000000000040428a in _start () at ../sysdeps/x86_64/start.S:120

/configure --verbose \
--enable-debug=-g \
--enable-parallel \
--with-fortran \
--with-hdf5="$HDF5" \
--enable-lfs \
--enable-cgnstools \
--disable-shared \

Reply:
Was it the intel compiler that had an issue with this line originally?

Reply:
Yes, it was generating this issue when reaching the function callback setting :
``
#0  0x00000000006b8aa8 in write_cgns_1_$FUNC.0.3 ()
#1  0x00007ffff7ac31c7 in cgi_error (format=0x1 <Address 0x1 out of bounds>)
    at /home/mickael/CGNS/src/cgns_error.c:38
#2  0x00007ffff7b02c6f in cg_set_file_type (file_type=1) at /home/mickael/CGNS/src/cgnslib.c:733
#3  0x00007ffff7b60108 in cg_configure_f_ (what=0x1, value=0x7ffff7db99c0 <cgns_error_mess>, 
    ier=0x7ffff7db99e2 <cgns_error_mess+34>) at /home/mickael/CGNS/src/cg_ftoc.c:192
#4  0x000000000040bcc1 in write_cgns_1 () at /home/mickael/CGNS/src/tests/cgwrite_f03.F90:373
#5  0x000000000040965e in main ()
#6  0x00007ffff5962c05 in __libc_start_main () from /lib64/libc.so.6
#7  0x0000000000409569 in _start ()
``

Reply:
(void *)(*(size_t *)value

was needed to pass properly the c_funloc

Reply:
Maybe separating CG_CONFIG_ERROR is needed

Reply:
Yes, looks like it.

Reply:
are you working on a push, or should I do it for what I have so far?

Reply:
You can do it with what you have so far.
I have not found any improvement to push right now.